### PR TITLE
DMP-58 Set up App Insights - 2nd Attempt

### PR DESCRIPTION
**Pull request by alancoleman**
…also updated in prep to push to new Azure resources.

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-58

### Change description ###
Work to add Azure Application Insights to the three microservices.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by jackmaloney — CHANGES_REQUESTED** [review]
This deletes a load of code that is needed. Please fix

File: infrastructure/main.tf [hunk 1]
```diff
@@ -13,3 +13,60 @@ locals {
...
+
+# Key Vault secrets
+
+resource "azurerm_key_vault_secret" "app_insights_connection_string" {
+  name = var.azurerm_key_vault_secret_conn_str
+  value = azurerm_application_insights.appinsights.connection_string
+  key_vault_id = module.key-vault.key_vault_id
+}
+
+resource "azurerm_key_vault_secret" "AZURE_APPINSGHTS_KEY" {
```
Comment (richardbull) [inline infrastructure/main.tf:68]: Is this typo intentional? AZURE_APPINSGHTS_KEY

Comment (msl8r) [reply infrastructure/main.tf:68]: should be lowercase to

Comment (alancoleman) [reply infrastructure/main.tf:68]: Good spot @richardbull  and @msl8r,
This line came in from draft-store, I was basing my approach on theirs. Sloppy of me not to spot it. Will correct. Thanks

https://github.com/hmcts/draft-store/blob/master/infrastructure/main.tf#L179

File: infrastructure/output.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+/*
+# Resource Group outputs
```
Comment (jackmaloney) [inline infrastructure/output.tf:3]: Why do you have this file if it's all commented out?

File: infrastructure/variables.tf [hunk 3]
```diff
@@ -10,12 +10,64 @@ variable "location" {
...
+
 variable "deployment_namespace" {
   default = ""
 }
+
 variable "common_tags" {
   type = map(string)
 }
+/*
+# Resource Group variables
```
Comment (jackmaloney) [inline infrastructure/variables.tf:26]: Probably don't need these if all commented out?

File: infrastructure/main.tf [hunk 4]
```diff
@@ -13,3 +13,60 @@ locals {
...
+# ... said variables were moved here due to a Jenkins objection
+module "key-vault" {
+  source              = "git@github.com:hmcts/cnp-module-key-vault?ref=master"
+  product             = "darts"
+  env                 = ""
+  tenant_id           = ""
+  object_id           = ""
+  resource_group_name = azurerm_resource_group.rg.name
+  # https://github.com/hmcts/devops-azure-ad/blob/master/users/prod_users.yml
+  product_group_name          = "Key Vault product group name"
```
Comment (jackmaloney) [inline infrastructure/main.tf:55]: You need to find out what should go in here. That's why you are getting 'Error: No group found matching specified filter (displayName eq 'Key Vault product group name' and securityEnabled eq true)' in your Jenkins build. Maybe ask Scott or if not Platops if they can help you here.

File: infrastructure/variables.tf (hunk 3 above)
Comment (alancoleman) [reply infrastructure/variables.tf:26]: Thanks @jackmaloney, as above, I'll remove everything that isn't necessary in a code clean up when we have something working.

File: infrastructure/main.tf [hunk 5]
```diff
@@ -13,3 +13,60 @@ locals {
...
+      application_type,
+    ]
+  }
+}
+
+# this key vault is created in every environment, but preview, being short-lived,
+# will use the aat one instead
+# This section includes hardcoded variables that were in variables.tf ...
+# ... said variables were moved here due to a Jenkins objection
+module "key-vault" {
```
Comment (jackmaloney) [inline infrastructure/main.tf]: Please follow the pattern on here: https://github.com/hmcts/cnp-module-key-vault/tree/master#usage

File: charts/darts-api/values.yaml [hunk 6]
```diff
@@ -4,9 +4,9 @@ java:
...
-#  keyVaults:
-#    darts:
-#      secrets:
-#        - name: AppInsightsInstrumentationKey
-#          alias: azure.application-insights.instrumentation-key
+  keyVaults:
+    darts:
+      secrets:
+        - name: AppInsightsInstrumentationKey
+        alias: azure.application-insights.instrumentation-key
```
Comment (jackmaloney) [inline charts/darts-api/values.yaml:11]: Incorrect formatting. Tab it right

**Comment by alancoleman** [comment]
@scott-robertson1 I think these features need to be in a feature block. https://github.com/hmcts/darts-api/blob/c54a718760fab267e9b4090e0f62935dac45334c/infrastructure/main.tf#LL42C1-L45C55

I'm getting a pipeline error. https://sds-build.platform.hmcts.net/blue/organizations/jenkins/HMCTS%2Fdarts-api/detail/PR-42/59/pipeline/219

Cheers

**Comment by alancoleman** [comment]
> @scott-robertson1 I think these features need to be in a feature block. https://github.com/hmcts/darts-api/blob/c54a718760fab267e9b4090e0f62935dac45334c/infrastructure/main.tf#LL42C1-L45C55
> 
> I'm getting a pipeline error. https://sds-build.platform.hmcts.net/blue/organizations/jenkins/HMCTS%2Fdarts-api/detail/PR-42/59/pipeline/219
> 
> Cheers

No, maybe not. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_provider_registration

**Comment by alancoleman** [comment]
> > @scott-robertson1 I think these features need to be in a feature block. https://github.com/hmcts/darts-api/blob/c54a718760fab267e9b4090e0f62935dac45334c/infrastructure/main.tf#LL42C1-L45C55
> > I'm getting a pipeline error. https://sds-build.platform.hmcts.net/blue/organizations/jenkins/HMCTS%2Fdarts-api/detail/PR-42/59/pipeline/219
> > Cheers
> 
> No, maybe not. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_provider_registration

https://discuss.hashicorp.com/t/at-least-1-features-blocks-are-required-issue/44511/2

**Comment by alancoleman** [comment]
@jackmaloney  and @davet1985, it looks as though we are at a point where we can now move forward with this. We Discussed this on Friday Dave. Once the vault secrets have been created on staging they'll be available for the PR, then we'll be able to check the PR With the references to those secrets exposed.

**Comment by davet1985** [comment]
> @jackmaloney and @davet1985, it looks as though we are at a point where we can now move forward with this. We Discussed this on Friday Dave. Once the vault secrets have been created on staging they'll be available for the PR, then we'll be able to check the PR With the references to those secrets exposed.

The branch needs updating and I think has some conflicts again. If you can sort that out I can review and we can potentially merge this part of it.

**Comment by alancoleman** [comment]
Morning @jackmaloney, branch has been updated. Let me know what the next stage of the process is. Thanks

**Pull request closed without merging**